src/anet.c

```diff
@@ -422,12 +423,20 @@ static int anetV6Only(char *err, int s) {
     return ANET_OK;
 }
 
-static int _anetTcpServer(char *err, int port, char *bindaddr, int af, int backlog)
+static int _anetTcpServer(char *err, int port, char *addr, int af, int backlog)
 {
     int s = -1, rv;
     char _port[6];  /* strlen("65535") */
     struct addrinfo hints, *servinfo, *p;
+    char* bindaddr = NULL;
+    char* dev = NULL;
 
+    if (addr != NULL) {
+        bindaddr = zstrdup(addr);
+        dev = strchr(bindaddr, '@');
```

Comment:
I'd make all of this based on whether we have `SO_BINDTODEVICE` or not, so it we don't support it we son't even try to look for `'@'` in the bind address.

Reply:
I think we should check it all the time. If the current OS doesn't support it, we should report it more accurately, rather than reporting the address is invalid.

Reply:
I agree in theory, the problem is that you're not checking against the **current OS** but against the **build OS**. So even if the current OS supports it, if it was built on another OS it'll fail. So unless you change the code to check if the current OS supports it I think it's better to either include the feature completely in the build or exclude it completely.

Reply:
Got it. Reasonable. And it is difficult to check if the current OS supports it.

Reply:
fixed